collections/nemo_asr/nemo_asr/parts/manifest.py

```diff
@@ -1,185 +1,78 @@
-            with open(manifest_path, "r", encoding="utf-8") as fh:
-                for line in fh:
-                    yield json.loads(line)
-
-    @staticmethod
-    def load_transcript(transcript_path):
-        with open(transcript_path, 'r', encoding="utf-8") as transcript_file:
-            transcript = transcript_file.read().replace('\n', '')
-        return transcript
-
-    @property
-    def duration(self):
-        return self._duration
-
-    @property
-    def filtered_duration(self):
-        return self._filtered_duration
-
-    @property
-    def data(self):
-        return list(self._data)
-
-
-class ManifestEN(ManifestBase):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-    @staticmethod
-    def normalize_text(text, labels):
-        # Punctuation to remove
-        punctuation = string.punctuation
-        # Define punctuation that will be handled by text cleaner
-        punctuation_to_replace = {
-            "+": "plus",
-            "&": "and",
-            "%": "percent"
-        }
-        for char in punctuation_to_replace:
-            punctuation = punctuation.replace(char, "")
-        # We might also want to consider:
-        # @ -> at
-        # -> number, pound, hashtag
-        # ~ -> tilde
-        # _ -> underscore
-
-        # If a punctuation symbol is inside our vocab, we do not remove
-        # from text
-        for l in labels:
-            punctuation = punctuation.replace(l, "")
-
-        # Turn all other punctuation to whitespace
-        table = str.maketrans(punctuation, " " * len(punctuation))
-
-        try:
-            text = clean_text(text, table, punctuation_to_replace)
-        except BaseException:
-            nemo.logging.warning("WARNING: Normalizing {} failed".format(text))
-            return None
-
-        return text
+from typing import Union, Iterator, Dict, Any, List
+
+
+def item_iter(
+    manifests_files: Union[str, List[str]]
+) -> Iterator[Dict[str, Any]]:
+    """Iterate through json lines of provided manifests.
+
+    NeMo ASR pipelines often assume certain manifest files structure. In
+    particular, each manifest files should consist of line per samples files w
+    with each line being correct json dict. Each such json dict should have a
+    field for audio file string, a field for duration float and a field for
+    text string.
+
+    Args:
+        manifests_files: Either single string file or list of such -
+            manifests to yield items from.
+
+    Yields:
+        Parsed key to value item dicts.
+
+    Raises:
+        ValueError: If met invalid json line structure.
+    """
+
+    if isinstance(manifests_files, str):
+        manifests_files = [manifests_files]
+
+    for manifest_file in manifests_files:
+        with open(manifest_file, 'r') as f:
+            for line in f:
+                item = __parse_item(line)
+
+                yield item
+
+
+def __parse_item(line: str) -> Dict[str, Any]:
```

Comment:
Let's also support offset

Reply:
An an non-mandatory field, that is None by default?

Reply:
Sure that works.

Reply:
Done.